sql/core/src/main/scala/org/apache/spark/sql/execution/command/tables.scala

```diff
@@ -58,18 +63,32 @@ case class CreateTableLikeCommand(
       throw new AnalysisException(
         s"Source table in CREATE TABLE LIKE does not exist: '$sourceTable'")
     }
-    if (catalog.isTemporaryTable(sourceTable)) {
-      throw new AnalysisException(
-        s"Source table in CREATE TABLE LIKE cannot be temporary: '$sourceTable'")
-    }
 
-    val tableToCreate = catalog.getTableMetadata(sourceTable).copy(
-      identifier = targetTable,
-      tableType = CatalogTableType.MANAGED,
-      createTime = System.currentTimeMillis,
-      lastAccessTime = -1).withNewStorage(locationUri = None)
+    val sourceTableDesc = catalog.getTableMetadata(sourceTable)
 
-    catalog.createTable(tableToCreate, ifNotExists)
+    val newSerdeProp =
+      if (DDLUtils.isDatasourceTable(sourceTableDesc)) {
+        val newPath = catalog.defaultTablePath(targetTable)
+        sourceTableDesc.storage.properties.filterKeys(_.toLowerCase != "path") ++
+          Map("path" -> newPath)
+      } else {
+        sourceTableDesc.storage.properties
+      }
+    val newStorage = sourceTableDesc.storage.copy(
+      locationUri = None,
+      properties = newSerdeProp)
+
+    val newTableDesc =
+      CatalogTable(
+        identifier = targetTable,
+        tableType = CatalogTableType.MANAGED,
+        storage = newStorage,
+        schema = sourceTableDesc.schema,
+        provider = sourceTableDesc.provider,
```

Comment:
Will fix it tomorrow. I need to recheck whether the new change is consistent with the behavior of Hive. Thanks!